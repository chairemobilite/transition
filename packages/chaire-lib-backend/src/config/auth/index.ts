/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
// Unused, but must be imported to make sure the environment is configured at this point, otherwise process.env will have undefined values
import _dotenv from '../dotenv.config'; // eslint-disable-line @typescript-eslint/no-unused-vars
import passport, { PassportStatic } from 'passport';
import GoogleStrategy from 'passport-google-oauth';
import FacebookStrategy from 'passport-facebook';
import url from 'url';
import localLogin from './localLogin.config';
import config from '../server.config';
import { IAuthModel, IUserModel } from '../../services/auth/authModel';

export default <U extends IUserModel>(authModel: IAuthModel<U>): PassportStatic => {
    if (process.env.GOOGLE_OAUTH_CLIENT_ID) {
        passport.use(
            new GoogleStrategy.OAuth2Strategy(
                {
                    clientID: process.env.GOOGLE_OAUTH_CLIENT_ID,
                    clientSecret: process.env.GOOGLE_OAUTH_SECRET_KEY,
                    callbackURL: new url.URL('/googleoauth', process.env.HOST).href
                },
                (accessToken, refreshToken, profile, done) => {
                    authModel
                        .find({ google_id: profile.id })
                        .then(async (user) => {
                            if (user !== undefined) {
                                // TODO Should sanitize user, but see if other fields need to be included
                                done(null, { ...user });
                            } else {
                                const newUser = await authModel.createAndSave({ googleId: profile.id, isTest: false });
                                done(null, newUser !== null ? newUser.sanitize() : false);
                            }
                            return null;
                        })
                        .catch(() => {
                            return done('Cannot verify google id credentials');
                        });
                }
            )
        );
    }

    if (process.env.FACEBOOK_APP_ID) {
        passport.use(
            new FacebookStrategy(
                {
                    clientID: process.env.FACEBOOK_APP_ID,
                    clientSecret: process.env.FACEBOOK_APP_SECRET,
                    callbackURL: new url.URL('/facebookoauth', process.env.HOST).href
                },
                (accessToken, refreshToken, profile, done) => {
                    authModel
                        .find({ facebook_id: profile.id })
                        .then(async (user) => {
                            if (user !== undefined) {
                                // TODO Should sanitize user, but see if other fields need to be included
                                done(null, { ...user });
                            } else {
                                const newUser = await authModel.createAndSave({
                                    facebookId: profile.id,
                                    isTest: false
                                });
                                done(null, newUser !== null ? newUser.sanitize() : false);
                            }
                            return null;
                        })
                        .catch(() => {
                            return done('Cannot verify facebook id credentials');
                        });
                }
            )
        );
    }

    if (!config.auth || config.auth.localLogin !== undefined || config.separateAdminLoginPage === true) {
        localLogin(passport, authModel);
    }
    if (config.auth && config.auth.passwordless !== undefined) {
        const passwordlessConfig = require('./passwordless.config');
        // FIXME It used to work without the next line, not anymore... probably some compilation issue
        const pwdlessConfig = passwordlessConfig.default ? passwordlessConfig.default : passwordlessConfig;
        pwdlessConfig(passport, authModel);
    }
    if (config.auth && config.auth.anonymous === true) {
        const anonymousLoginStrategy = require('../../services/auth/anonymousLoginStrategy');
        // FIXME It used to work without the next line, not anymore... probably some compilation issue
        const AnonymousLoginStrategy = anonymousLoginStrategy.default
            ? anonymousLoginStrategy.default
            : anonymousLoginStrategy;
        passport.use('anonymous-login', new AnonymousLoginStrategy(authModel));
    }

    // TODO user is Express.User type and does not have an id type
    passport.serializeUser((user: any, done) => {
        done(null, user.id);
        return null;
    });

    passport.deserializeUser(async (id, done) => {
        try {
            const model = await authModel.getById(id as number);
            if (model !== undefined) {
                done(null, model.attributes);
                return null;
            }
            done('User does not exist');
        } catch (error) {
            done(`Cannot deserialize user data, ${error}`);
        }
    });

    return passport;
};
