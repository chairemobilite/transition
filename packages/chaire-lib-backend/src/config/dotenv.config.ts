/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import path from 'path';
import fs from 'fs';

if (!process.env.NODE_ENV) {
    process.env.NODE_ENV = 'development';
}

// INIT_CWD is the directory from which the script was run (usually root of the
// repo), while PWD would be the chaire-lib-backend directory if run with `yarn
// start`
const workingDir = process.env.INIT_CWD || process.env.PWD;

// Get the .env file. The return file path exists, the .env file can come from
// the TRANSITION_DOTENV environment variable, or is expected to be at the root
// of the directory
const getDotEnvFile = (): string | null => {
    // First, check the PROJECT_CONFIG environment variable
    if (typeof process.env.TRANSITION_DOTENV === 'string') {
        const dotEnvFile = process.env.TRANSITION_DOTENV.startsWith('/')
            ? process.env.TRANSITION_DOTENV
            : `${workingDir}/${process.env.TRANSITION_DOTENV}`;
        if (fs.existsSync(dotEnvFile)) {
            return path.normalize(dotEnvFile);
        }
        console.log(`.env file specified by TRANSITION_DOTENV environment variable does not exist: ${dotEnvFile}`);
    }
    const defaultDotEnv = `${workingDir}/.env`;
    if (fs.existsSync(defaultDotEnv)) {
        return defaultDotEnv;
    }
    console.log(
        `.env file not found. You can set the TRANSITION_DOTENV environment variable to point to the .env file. The default path is ${defaultDotEnv}`
    );
    return null;
};

const dotEnvPath = getDotEnvFile();
if (dotEnvPath !== null) {
    console.log(`Using .env file from ${dotEnvPath}`);
}

/* eslint-disable-next-line */
export default require('dotenv').config({ path: dotEnvPath });
