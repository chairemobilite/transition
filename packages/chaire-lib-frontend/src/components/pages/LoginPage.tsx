/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';
import { connect } from 'react-redux';
import { withTranslation, WithTranslation } from 'react-i18next';
import { History, Location } from 'history';
import appConfiguration from '../../config/application.config';
import LoginForm from '../forms/auth/localLogin/LoginForm';

export interface LoginPageProps {
    isAuthenticated?: boolean;
    history: History;
    location: Location;
    config: {
        allowRegistration?: boolean;
        forgotPasswordPage?: boolean;
        auth?: {
            localLogin: {
                allowRegistration?: boolean;
                forgotPasswordPage?: boolean;
            };
        };
    };
    login?: boolean;
}

export const LoginPage: React.FunctionComponent<LoginPageProps & WithTranslation> = (
    props: LoginPageProps & WithTranslation
) => {
    React.useEffect(() => {
        if (props.isAuthenticated) {
            props.history.push(appConfiguration.homePage);
        }
    }, []);

    const allowRegistration =
        props.config.allowRegistration !== false && props.config.auth?.localLogin?.allowRegistration !== false;

    return (
        <React.Fragment>
            <LoginForm
                history={props.history}
                location={props.location}
                withForgotPassword={
                    props.config.forgotPasswordPage === true ||
                    props.config.auth?.localLogin?.forgotPasswordPage === true
                }
            />
            <div className="apptr__separator-medium"></div>
            <div className="apptr__footer-link-container">
                {allowRegistration && (
                    <a className="register-link _oblique" href="/register">
                        {props.t('auth:registerIfYouHaveNoAccount')}
                    </a>
                )}
            </div>
        </React.Fragment>
    );
};

const mapStateToProps = (state) => {
    return { isAuthenticated: state.auth.isAuthenticated, login: state.auth.login };
};

export default connect(mapStateToProps)(withTranslation('auth')(LoginPage));
