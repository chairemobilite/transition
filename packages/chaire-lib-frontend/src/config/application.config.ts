/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import { TFunction } from 'i18next';
import { UserPages } from 'chaire-lib-common/lib/services/user/userType';
import { MouseEventHandler } from 'react';

export type UserMenuItem = {
    getText: (t: TFunction) => string;
    action: MouseEventHandler;
    /**
     * Extra action to run after the action has been run. This allows to trigger
     * some events on the page. 'refreshLang' will cause a refresh of the
     * translation strings on the page.
     */
    postExec?: 'refreshLang';
    confirmModal?: {
        title: (t: TFunction) => string;
        label: (t: TFunction) => string;
    };
};

export type ApplicationConfiguration<AdditionalConfig> = {
    /**
     * Main page to redirect to by default when logging in
     */
    homePage: string;
    /**
     * Pages available to the user, once logged in
     */
    pages: UserPages[];
    /**
     * Items that will be added to the menu, when clicking on the logged in user
     * name
     */
    userMenuItems: UserMenuItem[];
} & AdditionalConfig;

const appConfiguration: ApplicationConfiguration<any> = {
    homePage: '/',
    pages: [],
    userMenuItems: []
};

export const setApplicationConfiguration = <AdditionalConfig>(
    config: Partial<ApplicationConfiguration<AdditionalConfig>>
) => Object.keys(config).forEach((configKey) => (appConfiguration[configKey] = config[configKey]));

export default appConfiguration;
