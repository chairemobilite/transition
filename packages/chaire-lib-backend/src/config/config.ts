/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import path from 'path';
import os from 'os';
import _cloneDeep from 'lodash.clonedeep';

import config, { setProjectConfiguration } from 'chaire-lib-common/lib/config/shared/project.config';
import fs from 'fs';

export type BaseServerConfig = {
    /**
     * Absolute directory where project data will be stored (import files, osrm data, user data, caches, etc)
     */
    projectDirectory: string;
    /**
     * The default disk usage quota for a user. The string is a number suffixed
     * with the units, kb, mb or gb
     */
    userDiskQuota: string;
    /**
     * Maximum file upload size, in megabytes. Defaults to 256MB
     */
    maxFileUploadMB: number;
    maxParallelCalculators: number;
};

export type ServerConfiguration<AdditionalServerConfig> = BaseServerConfig & AdditionalServerConfig;

// Initialize default server side configuration
export const serverConfig: ServerConfiguration<unknown> = {
    userDiskQuota: '1gb',
    maxFileUploadMB: 256,
    maxParallelCalculators: 1,
    // Initialize runtime directory at the root of the repo
    projectDirectory: path.normalize(`${__dirname}/../../../../runtime/`)
};

/**
 * Set the server configuration options.
 *
 * @param config The configuration options to set
 * @returns
 */
export const setServerConfiguration = <AdditionalServerConfig>(
    config: Partial<ServerConfiguration<AdditionalServerConfig>>
) => Object.keys(config).forEach((configKey) => (serverConfig[configKey] = config[configKey]));

const etcConfigFile = '/etc/transition/config.js';
const homeDirConfigFileRelative = '.config/transition/config.js';

// INIT_CWD is the directory from which the script was run (usually root of the
// repo), while PWD would be the chaire-lib-backend directory if run with `yarn
// start`
const workingDir = process.env.INIT_CWD || process.env.PWD;

// Get the config file. The return file path exists, the configuration file can
// come from the PROJECT_CONFIG environment variable, then
// $HOME/.config/transition/config.js, then /etc/transition/config.js
const getConfigFilePath = (): string => {
    // First, check the PROJECT_CONFIG environment variable
    if (typeof process.env.PROJECT_CONFIG === 'string') {
        const projectConfigFile = process.env.PROJECT_CONFIG.startsWith('/')
            ? process.env.PROJECT_CONFIG
            : `${workingDir}/${process.env.PROJECT_CONFIG}`;
        if (fs.existsSync(projectConfigFile)) {
            return path.normalize(projectConfigFile);
        }
        console.log(
            `Configuration file specified by PROJECT_CONFIG environment variable does not exist: ${projectConfigFile}`
        );
    }
    const homeDirConfigFile = path.normalize(`${os.homedir()}/${homeDirConfigFileRelative}`);
    if (fs.existsSync(homeDirConfigFile)) {
        return homeDirConfigFile;
    }
    if (fs.existsSync(etcConfigFile)) {
        return etcConfigFile;
    }
    throw `Configuration file not found. Either set the PROJECT_CONFIG environment variable to point to the config file, or use files ${homeDirConfigFile} or ${etcConfigFile} paths`;
};

const parseServerConfig = (configFromFile: { [key: string]: any }): Partial<BaseServerConfig> => {
    // Split server configuration from other configuration
    const serverConfigFromFile: Partial<BaseServerConfig> = {};

    // Extract the server configuration from the rest of the config
    // TODO Validate the fields individually before setting them
    const fieldsForServer = ['projectDirectory', 'userDiskQuota', 'maxFileUploadMB', 'maxParallelCalculators'];
    fieldsForServer.forEach((field) => {
        if (configFromFile[field] !== undefined) {
            serverConfigFromFile[field] = configFromFile[field];
            delete configFromFile[field];
        }
    });

    // Default project directory is `runtime` at the root of the repo
    if (serverConfigFromFile.projectDirectory === undefined && configFromFile.projectShortname !== undefined) {
        serverConfigFromFile.projectDirectory = path.normalize(
            `${__dirname}/../../../../runtime/${configFromFile.projectShortname}`
        );
    }
    return serverConfigFromFile;
};

let configInitialized = false;

export const initializeConfig = <ServerC, ProjectC>(
    parseServer?: (configFromFile: { [key: string]: any }) => ServerC,
    parseProject?: (configFromFile: { [key: string]: any }) => ProjectC,
    forceReload = false
) => {
    if (configInitialized === true && !forceReload) {
        return;
    }
    const configFileNormalized = getConfigFilePath();

    try {
        const configFileContent = require(configFileNormalized);
        const configFromFile = _cloneDeep(configFileContent);
        console.log(`Read server configuration from ${configFileNormalized}`);

        const serverConfigFromFile = parseServerConfig(configFromFile);

        const appServerconfig = parseServer !== undefined ? parseServer(configFromFile) : {};
        const appProjectConfig = parseProject !== undefined ? parseProject(configFromFile) : {};
        // TODO Validate the serverConfigFromFile and configFromFile objects before setting it
        setServerConfiguration<ServerC>(
            Object.assign({}, serverConfigFromFile, appServerconfig) as ServerConfiguration<ServerC>
        );
        setProjectConfiguration<ProjectC>(Object.assign({}, configFromFile, appProjectConfig));
        configInitialized = true;
    } catch (error) {
        console.error(`Error loading server configuration in file ${configFileNormalized}`);
    }
};

/** Application configuration, does not include the server configuration */
export const projectConfig = config;
