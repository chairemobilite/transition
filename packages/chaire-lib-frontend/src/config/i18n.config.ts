/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import moment from 'moment-business-days';
import LanguageDetector from 'i18next-browser-languagedetector';
import HttpApi from 'i18next-http-backend';

import config from 'chaire-lib-common/lib/config/shared/project.config';

const detectorOrder = config.detectLanguage ? ['cookie', 'localStorage', 'navigator'] : ['cookie', 'localStorage'];

i18n.use(LanguageDetector)
    .use(initReactI18next)
    .use(HttpApi)
    .init(
        {
            detection: {
                // order and from where user language should be detected
                order: detectorOrder,
                caches: ['localStorage', 'cookie']
            },
            load: 'languageOnly', // no region-specific,
            preload: config.languages,
            whitelist: config.languages,
            nonExplicitSupportedLngs: false,
            fallbackLng: config.defaultLocale || 'en',
            debug: false,
            interpolation: {
                escapeValue: false // not needed for react!!
            },
            react: {
                wait: true,
                useSuspense: false
            }
        },
        (err, _t) => {
            if (err) {
                console.log(err);
            }
        }
    );

if (i18n.language) {
    i18n.changeLanguage(i18n.language.split('-')[0]); // force remove region specific
}

if (!i18n.language || config.languages.indexOf(i18n.language) <= -1) {
    i18n.changeLanguage(config.defaultLocale);
}

i18n.on('languageChanged', (language) => {
    document.documentElement.setAttribute('lang', language);
    moment.locale(language);
});
export default i18n;
