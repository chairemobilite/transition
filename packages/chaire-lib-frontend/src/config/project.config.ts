import config, {
    ProjectConfiguration,
    setProjectConfiguration
} from 'chaire-lib-common/lib/config/shared/project.config';
declare let __CONFIG__: ProjectConfiguration<unknown>;

const frontendConfig = __CONFIG__;
if (frontendConfig === undefined) {
    console.error('__CONFIG__ global variable is not set. Webpack should define it');
} else {
    setProjectConfiguration(frontendConfig);
}

export default config;
