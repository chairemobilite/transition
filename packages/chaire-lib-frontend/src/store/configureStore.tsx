/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import { createStore, combineReducers, applyMiddleware, compose, Store } from 'redux';
import { thunk, ThunkMiddleware } from 'redux-thunk';

import { authReducer } from './auth';
import { AuthState } from './auth/types';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const initialState = {};

export type RootState = {
    auth: AuthState;
};

export default (preloadedState = initialState): Store => {
    const store = createStore(
        combineReducers({
            auth: authReducer
        }),
        preloadedState,
        composeEnhancers(applyMiddleware(thunk as ThunkMiddleware<RootState>))
    );
    return store;
};
