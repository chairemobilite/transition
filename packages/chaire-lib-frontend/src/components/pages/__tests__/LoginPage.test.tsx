/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
*/
import * as React from 'react';

import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { MemoryRouter, Routes, Route } from 'react-router';
import { setApplicationConfiguration } from '../../../config/application.config';
import LoginPage from '../LoginPage';

const mockHomePage = '/Homepage';
setApplicationConfiguration({ homePage: mockHomePage });

import configureStore from '../../../store/configureStore';  // Create similar to docs example

const renderWithProviders = (
    ui,
    {
        preloadedState = {
            auth: {
                isAuthenticated: false
            }
        },
        initialEntries = ['/login'],
        ...renderOptions
    } = {}
) => {
    const store = configureStore(preloadedState);
    const Wrapper = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={initialEntries}>
                <Routes>
                    <Route path="/login" element={children} />
                    <Route path={mockHomePage} element={<div>Home Page</div>} />
                    <Route path="/register" element={<div>Register Page</div>} />
                    <Route path="/forgot" element={<div>Forgot Password Page</div>} />
                </Routes>
            </MemoryRouter>
        </Provider>
    );

    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

describe('Login page rendering', () => {
    test('Is already authenticated, redirect to home page', () => {

        renderWithProviders(
            <LoginPage config={{
                allowRegistration: true,
                forgotPasswordPage: true
            }}/>,
            { preloadedState: { auth: { isAuthenticated: true } } }
        );

        expect(screen.getByText('Home Page')).toBeInTheDocument();
    });

    test('Allow registration, no forgot password page', () => {
        const { container } = renderWithProviders(
            <LoginPage config={{
                allowRegistration: true,
                forgotPasswordPage: false
            }}/>
        );
        expect(container).toMatchSnapshot();
    });

    test('Registration not allowed, no forgot password page', () => {
        const { container } = renderWithProviders(
            <LoginPage config={{
                allowRegistration: false
            }}/>
        );
        expect(container).toMatchSnapshot();
    });

    test('Registration allowed, allow forget password', () => {
        const { container } = renderWithProviders(
            <LoginPage config={{
                allowRegistration: true,
                forgotPasswordPage: true
            }}/>
        );
        expect(container).toMatchSnapshot();
    });
});
