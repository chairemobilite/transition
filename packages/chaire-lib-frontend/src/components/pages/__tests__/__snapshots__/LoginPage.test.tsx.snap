// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page rendering Allow registration, no forgot password page 1`] = `
<div>
  <form
    class="apptr__form apptr__form-auth"
  >
    <div
      class="apptr__form-label-container center"
    >
      <div
        class="apptr__form__label-standalone no-question"
      >
        <p>
          auth:pleaseEnterLoginCredentials
        </p>
      </div>
    </div>
    <div
      class="apptr__form-container question-empty"
    >
      <div
        class="apptr__form-input-container"
      >
        <label
          class="_flex"
          for="usernameOrEmail"
        >
          auth:UsernameOrEmail
        </label>
        <input
          class="apptr__form-input apptr__form-input-string apptr__input apptr__input-string"
          id="usernameOrEmail"
          name="usernameOrEmail"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="apptr__form-container question-empty"
    >
      <div
        class="apptr__form-input-container"
      >
        <label
          class="_flex"
          for="password"
        >
          auth:Password
        </label>
        <input
          class="apptr__form-input apptr__form-input-string apptr__input apptr__input-string"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="center"
    >
      <button
        class="button green large"
        type="button"
      >
         
        auth:Login
      </button>
    </div>
  </form>
  <div
    class="apptr__separator-medium"
  />
  <div
    class="apptr__footer-link-container"
  >
    <a
      class="register-link _oblique"
      href="/register"
    >
      auth:registerIfYouHaveNoAccount
    </a>
  </div>
</div>
`;

exports[`Login page rendering Registration allowed, allow forget password 1`] = `
<div>
  <form
    class="apptr__form apptr__form-auth"
  >
    <div
      class="apptr__form-label-container center"
    >
      <div
        class="apptr__form__label-standalone no-question"
      >
        <p>
          auth:pleaseEnterLoginCredentials
        </p>
      </div>
    </div>
    <div
      class="apptr__form-container question-empty"
    >
      <div
        class="apptr__form-input-container"
      >
        <label
          class="_flex"
          for="usernameOrEmail"
        >
          auth:UsernameOrEmail
        </label>
        <input
          class="apptr__form-input apptr__form-input-string apptr__input apptr__input-string"
          id="usernameOrEmail"
          name="usernameOrEmail"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="apptr__form-container question-empty"
    >
      <div
        class="apptr__form-input-container"
      >
        <label
          class="_flex"
          for="password"
        >
          auth:Password
        </label>
        <input
          class="apptr__form-input apptr__form-input-string apptr__input apptr__input-string"
          id="password"
          name="password"
          type="password"
          value=""
        />
        <div
          class="right _small"
        >
          <a
            class="_oblique"
            data-discover="true"
            href="/forgot"
          >
            auth:forgotPassword
          </a>
        </div>
      </div>
    </div>
    <div
      class="center"
    >
      <button
        class="button green large"
        type="button"
      >
         
        auth:Login
      </button>
    </div>
  </form>
  <div
    class="apptr__separator-medium"
  />
  <div
    class="apptr__footer-link-container"
  >
    <a
      class="register-link _oblique"
      href="/register"
    >
      auth:registerIfYouHaveNoAccount
    </a>
  </div>
</div>
`;

exports[`Login page rendering Registration not allowed, no forgot password page 1`] = `
<div>
  <form
    class="apptr__form apptr__form-auth"
  >
    <div
      class="apptr__form-label-container center"
    >
      <div
        class="apptr__form__label-standalone no-question"
      >
        <p>
          auth:pleaseEnterLoginCredentials
        </p>
      </div>
    </div>
    <div
      class="apptr__form-container question-empty"
    >
      <div
        class="apptr__form-input-container"
      >
        <label
          class="_flex"
          for="usernameOrEmail"
        >
          auth:UsernameOrEmail
        </label>
        <input
          class="apptr__form-input apptr__form-input-string apptr__input apptr__input-string"
          id="usernameOrEmail"
          name="usernameOrEmail"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="apptr__form-container question-empty"
    >
      <div
        class="apptr__form-input-container"
      >
        <label
          class="_flex"
          for="password"
        >
          auth:Password
        </label>
        <input
          class="apptr__form-input apptr__form-input-string apptr__input apptr__input-string"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="center"
    >
      <button
        class="button green large"
        type="button"
      >
         
        auth:Login
      </button>
    </div>
  </form>
  <div
    class="apptr__separator-medium"
  />
  <div
    class="apptr__footer-link-container"
  />
</div>
`;
